functions/__sf_section_dir.fish

```diff
@@ -27,11 +27,12 @@ function __sf_section_dir -d "Display the current truncated directory"
 
 	set -l dir
 	set -l tmp
+	set -l resolvedPwd (pwd -P 2>/dev/null)
```

Comment:
For easy compatibility with 2.7.0, put `; or pwd` in there, because that didn't have the `-P` option and will error out.

Reply:
Unfortunately, this causes a bug on 2.7.0 due to `pwd` being the logical location while the git root is a physical location.
It seems to work consistently now when using `command pwd -P`. Do you see any drawbacks to this approach?

Reply:
>Unfortunately, this causes a bug on 2.7.0 due to pwd being the logical location while the git root is a physical location.

In 2.7.0, the pwd _is_ the physical location, always.

So `pwd` prints the same thing in 2.7.0, as `pwd -P` does in 3.0.

Using just `pwd` causes a bug in **3.0**, not 2.7.

Just to be clear, my proposal is:

```fish
set -l resolvedPWD (pwd -P 2>/dev/null; or pwd)
```

which will try `pwd -P`, which will fail if the option isn't supported, and then it'll run `pwd`.

>It seems to work consistently now when using command pwd -P. Do you see any drawbacks to this approach?

It's an additional fork (`command pwd` takes a millisecond or so on my system, hot-cache, while the builtin is unmeasurable). Plus, it's an additional external command you require.

pwd (with -P) is a POSIX requirement, but still.

Not something that's hugely important.

Reply:
Excellent! Yes, you're right. It was 3.0 that caused the bug.
Thank you for the clarification. 😄